modules/protocol/src/vector.ts

```diff
@@ -263,50 +257,117 @@ export class Vector implements IVectorProtocol {
+
+        // First, update the highest seen nonce
+        this.highestNonce[received.update.channelAddress] = received.update.nonce;
+
+        // Apply the update at the higher nonce
+        const params = getParamsFromUpdate(received.update);
+        if (params.isError) {
+          // TODO: respond here so initiator doesnt just time out
           this.logger.warn(
-            { method, methodId, error: jsonifyError(inboundRes.getError()!) },
-            "Failed to apply inbound update",
+            { method, methodId, error: jsonifyError(params.getError()!) },
+            "Could not get params from update",
           );
           return;
         }
+        const result = await this.addToQueue(params.getValue(), received.previousUpdate);
+        if (result.isError) {
+          this.logger.warn({ method, methodId, error: jsonifyError(result.getError()!) }, "Failed to apply update");
+          // reset to previous nonce here while nonces cant be burned
+          // TODO: implement burned nonces
+          this.highestNonce[received.update.channelAddress] = highestChannelNonce;
+          return;
+        }
 
-        const { updatedChannel, updatedActiveTransfers, updatedTransfer } = inboundRes.getValue();
-
-        // TODO: more efficient dispute events
-        // // If it is setup, watch for dispute events in channel
-        // if (received.update.type === UpdateType.setup) {
-        //   this.logger.info({ channelAddress: updatedChannel.channelAddress }, "Registering channel for dispute events");
-        //   const registrationRes = await this.chainReader.registerChannel(
-        //     updatedChannel.channelAddress,
-        //     updatedChannel.networkContext.chainId,
-        //   );
-        //   if (registrationRes.isError) {
-        //     this.logger.warn(
-        //       { ...jsonifyError(registrationRes.getError()!) },
-        //       "Failed to register channel for dispute watching",
-        //     );
-        //   }
-        // }
+        const { updatedChannel, updatedActiveTransfers, updatedTransfer } = result.getValue();
 
         this.evts[ProtocolEventName.CHANNEL_UPDATE_EVENT].post({
           updatedChannelState: updatedChannel,
           updatedTransfers: updatedActiveTransfers,
           updatedTransfer,
         });
-        this.logger.debug({ method, methodId }, "Method complete");
+        this.logger.debug({ method, methodId, channelNonce: updatedChannel.nonce }, "Applied received update");
+
+        // Check if you are currently proposing an update
+        const proposed = this.inProgressUpdate[received.update.channelAddress];
+
+        // If no, return
+        if (!proposed) {
+          return;
+        }
+
+        // If so, check if the update includes our proposed update
+        const proposedTransfer =
+          proposed.type === UpdateType.create ? getTransferFromUpdate(proposed, updatedChannel) : undefined;
+        const included = this.includesOurProposedUpdate(
```

Comment:
There may be a wrinkle here to consider.

In this scenario where the proposed update is included we have to realize that we may not have the double-signed state for our proposed update. What if their proposed update is rejected? Do the participants disagree on the latest double-signed state?